aiida/orm/autogroup.py

```diff
@@ -7,173 +7,286 @@
+
+    @staticmethod
+    def _matches(string, filter_string):
+        """Check if 'string' matches the 'filter_string' (used for include and exclude filters).
+
+        If 'filter_string' does not end with a % sign, perform an exact match.
+        Otherwise, strip the '%' sign and match with string.startswith(filter_string[:-1]).
+
+        :param string: the string to match.
+        :param filter_string: the filter string.
         """
-        if not isinstance(gname, str):
-            raise exceptions.ValidationError('group name must be a string')
-        self.group_name = gname
+        if filter_string.endswith('%'):
+            return string.startswith(filter_string[:-1])
+        return string == filter_string
 
-    def is_to_be_grouped(self, the_class):
+    def is_to_be_grouped(self, node):
         """
-        Return whether the given class has to be included in the autogroup according to include/exclude list
+        Return whether the given node has to be included in the autogroup according to include/exclude list
 
-        :return (bool): True if the_class is to be included in the autogroup
+        :return (bool): True if ``node`` is to be included in the autogroup
         """
+        # I import here to avoid circular imports
+        from aiida.orm.nodes.process import ProcessNode
+
+        # strings, including possibly 'all'
         include = self.get_include()
-        include_ws = self.get_include_with_subclasses()
-        if (('all.' in include) or
-                (the_class._plugin_type_string in include) or
-                any([the_class._plugin_type_string.startswith(i) for i in include_ws])
-        ):
-            exclude = self.get_exclude()
-            exclude_ws = self.get_exclude_with_subclasses()
-            if ((not 'all.' in exclude) or
-                    (the_class._plugin_type_string in exclude) or
-                    any([the_class._plugin_type_string.startswith(i) for i in exclude_ws])
-            ):
-                return True
-            else:
-                return False
+        exclude = self.get_exclude()
+        if include is None and exclude is None:
+            # Include all classes by default if nothing is explicitly specified.
+            return True
+        if include is not None and exclude is not None:
+            # We should never be here, anyway - this should be catched by the `set_include/exclude` methods
+            raise ValueError("You cannot specify both an 'include' and an 'exclude' list")
+
+        the_class = node.__class__
+        if issubclass(the_class, ProcessNode):
+            try:
+                the_class = node.process_class
+            except ValueError:
+                # It does not have a process class - we just check the node class then, it could be e.g.
+                # a bare CalculationNode.
+                pass
+        class_entry_point_string = get_entry_point_string_from_class(the_class.__module__, the_class.__name__)
```

Comment:
All of this can just be replaced with:
```
entry_point_string = node.process_type
# If there is no `process_type` we are dealing with a `Data` node so we get the entry point from the class
if not entry_point_string:
    entry_point_string = get_entry_point_string_from_class(node.__class__.__module__, node.__class__.__name__)
```

Reply:
Thanks